symengine/derivative.cpp

```diff
@@ -324,6 +323,12 @@ class DiffImplementation
         return fdiff(self, x);
     }
 
+    static RCP<const Basic> diff(const Dirichlet_eta &self,
```

Comment:
Can you try making this `OneArgFunction` instead of `Dirichlet_eta` then? If that doesn't work, I'll merge this